### DISCUSSION TITLE
Anyone who can help me with "fitView"?

### DISCUSSION THREAD

**Discussion by piyusht1402** [xyflow/xyflow#3079] [General]
I have made a reactflow app, a part of it is where user can toggle the layout view, i.e. horizontal or vertical :

This is my code-
  const toggleLayoutDirection = () => {
    setShowEdges(false);
    setZoomOut(true); // Trigger the zoom out animation
    setTimeout(() => {
      const newDirection = layoutDirection === 'BT' ? 'RL' : 'BT';
      setLayoutDirection(newDirection);
      const { nodes: newNodes, edges: newEdges } = getLayoutedElements(
        nodes,
        edges,
        newDirection
      );
      setNodes([...newNodes]);
      setEdges([...newEdges]);
      setZoomClass('');
      setZoomOut(false); // Reset the zoomOut state after the animation is complete
      setShowEdges(true);
    }, 200);
  };

Now i want to use fitView, so that whenever  user presses toggle layout button, the layout should not be larger than the available screen, (which is happening right now), The layout should fit the screen on clicking the toggle Layout button,  So, how can i add the fitView in my togglelayout function to get this done?

**Comment by bcakmakoglu** [comment]
Assuming you have wrapped your Flow component using `ReactFlowProvider` you can use something like this

```ts
const { fitView } = useReactFlow();

const toggleLayoutDirection = () => {
	// your code to layout

	fitView();
}
```

**Reply by piyusht1402** [reply]
Thankyou, this worked!

**Comment by vasanthid** [comment]
@piyusht1402 - I need to implement a similar use case. Would you mind sharing the implementation of 'getLayoutedElements' ?  I was looking at the examples using dagre and elkjs but was wondering if there is a way to do it without using those libraries.
